### ISSUE TITLE
Add new SR Types ext4, xfs and the old Type file

### ISSUE THREAD

**Issue by cocoon** [xcp-ng/xenadmin#130]
**Describe the bug**
XCP-NG will allow to use ext4 and XFS, details here:
https://xcp-ng.org/forum/topic/839/new-ext4-driver-please-test-it/9

XCP-NG Center needs to be made aware about it.

**To Reproduce**

yum install sm-additional-drivers --enablerepo='xcp-ng-updates_testing'
xe-toolstack-restart

create SR:
xe sr-create type=ext4 name-label=mySR host-uuid=<host_UUID> device-config:device=/dev/sdX

**Expected behavior**
If ext4-SR is selected it should show ext4 as Type (and of course xfs for XFS)

**Screenshots / Logfiles**

**XCP-ng Center:**
 - XCP-ng Center Version: 7.6.1
 - Desktop OS: Win10

**XCP-ng Server**
 - Hardware: VM
 - XCP-ng Version: 7.6

**Additional context**

Had a quick look and identified the following files where changes might be needed:

https://github.com/xcp-ng/xenadmin/blob/master/XenModel/XenAPI-Extensions/SR.cs#L50

**CanCreateWithXenCenter is for example checked by Detach/Reattach/Forget/Destroy SR**
https://github.com/xcp-ng/xenadmin/blob/master/XenModel/XenAPI-Extensions/SR.cs#L235

https://github.com/xcp-ng/xenadmin/blob/master/XenModel/XenAPI-Extensions/SR.cs#L250

**Friendly names from resources:**
https://github.com/xcp-ng/xenadmin/blob/master/XenModel/FriendlyNames.Designer.cs#L2694

https://github.com/xcp-ng/xenadmin/blob/master/XenModel/FriendlyNames.resx#L357
https://github.com/xcp-ng/xenadmin/blob/master/XenModel/FriendlyNames.ja.resx#L357
https://github.com/xcp-ng/xenadmin/blob/master/XenModel/FriendlyNames.zh-CN.resx#L357

And don't know if any wizards should support it ...

**Comment by cocoon** [comment]
https://github.com/xcp-ng/xenadmin/pull/131

**Comment by cocoon** [comment]
** Ext4 SR Detach/Repair**

![001-xcp-ng-ext4](https://user-images.githubusercontent.com/1071741/51436527-57c5e980-1c8f-11e9-8510-6ed4f75a5699.png)



**Destroy**

Destroying ext4 SR is currently not working by xapi:

```
Jan 20 08:33:45 localhost xapi: [debug|localhost|732 |Async.SR.destroy R:031eef19f33f|xapi] Unmarking SR after SR.destroy (task=OpaqueRef:031eef19-f33f-4d99-943b-f179cddce76c)
Jan 20 08:33:45 localhost xapi: [error|localhost|732 ||backtrace] Async.SR.destroy R:031eef19f33f failed with exception Server_error(SR_BACKEND_FAILURE_202, [ ; General backend error [opterr=Unsupported SR type: ext4];  ])
Jan 20 08:33:45 localhost xapi: [error|localhost|732 ||backtrace] Raised Server_error(SR_BACKEND_FAILURE_202, [ ; General backend error [opterr=Unsupported SR type: ext4];  ])

```

**Comment by borzel** [comment]
@cocoon please continue development here: https://github.com/xcp-ng/xenadmin/tree/feature-xcp-7.6-ext4-xfs
Your original PR was based on master which has not the right state for development ... I still need to figure a system out, how we continue with XCP-ng Center 7.7 or 8.0 ...

**Comment by borzel** [comment]
I'll try to add also the file SR, which is currently used with (experimental) ZFS datasets.

**Comment by cocoon** [comment]
> @cocoon I would **NOT** add ext4 and xfs to the CanCreateWithXenCenter() ... because we can't create it with XCP-ng Center.

Like mentioned above, it is not only used for creating SR, this list is checked for other features like detaching/forget/destroy/... the SR. If it is not in that list, you cannot detach the SR.

See here: 
https://github.com/xenserver/xenadmin/search?q=CanCreateWithXenCenter&type=Code
(I had to use the xenserver/xenadmin repository, because I got an error, that forks can not be searched currently)


The question is, would it be good to be able to detach ext4 SR? Because we currently say it is local, I see only one usage for it. If you have an additional Disk attached to a virtual XCP-NG, you could detach it and attach it to another one .. for whatever reasons.
Don't know if there is any additional benefit / usecase for it?

I find the check

**Comment by borzel** [comment]
hhmm... hmm... maybe :-/ hmm....

**Comment by cocoon** [comment]
OK let's remove it from the feature, I really don't know if it is useful or not. But it is easy to add again if it's needed or usable from a Wizard, so just remove it from there.

Maybe it would be useful to detach an ext4 to do maintenance / fsck or resize it ... but as all of this currently has to be done on the cli it makes not much difference.

**Comment by borzel** [comment]
Thats what I was thinking :-)

**Comment by cocoon** [comment]
I changed it here: https://github.com/xcp-ng/xenadmin/commit/c9a559e1351630ea3bffad2320cca58d2b15a4f6

**Comment by borzel** [comment]
@cocoon ok

I added XFS to IsLocalSR(): https://github.com/xcp-ng/xenadmin/commit/62757bd6b44f9d3389d077b3723471bf51af1b0b

**Comment by borzel** [comment]
merged to XCP-7.6 Relase Branch
